sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -559,6 +559,25 @@ case class DataSource(
       }
       globPath
     }.toSeq
+
+    if (checkFilesExist) {
```

Comment:
Do you need to remove the check and exception a few lines above then? It would fail if any path didn't have some files. (Also feel free to fix the indentation from line 549-558 above)

Reply:
I am not sure, but irregular indentations seems to be due to GitHub preview CSS.

Reply:
No need, I fixed it.

Reply:
Yes I mean line 557. I guess we can keep that because, overall, we are trying to throw `AnalysisException` in more cases, not fewer. Before, if one of several glob paths matched no files at all (underscore or not) it would throw. OK, that behavior we can keep, I guess, or at least that's a separate question.

Disregard this; I think it is OK.